jbpm-flow-builder/src/main/java/org/jbpm/compiler/ProcessBuilderImpl.java

```diff
@@ -426,7 +438,7 @@ private String createSplitRule(Process process,
         		connection.getToType() + "\"  @Propagation(EAGER) \n" +
         	"      ruleflow-group \"DROOLS_SYSTEM\" \n" +
         	"    when \n" +
-        	"      " + constraint + "\n" +
+        	"      " + constraint + getEntryPoint(constraint, process) +  "\n" +
```

Comment:
It's only kbase separation as process instance is still mainly based on runtime strategy so I'd say it's an antipattern